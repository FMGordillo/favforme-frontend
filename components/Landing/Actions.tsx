import { FunctionComponent } from "react";

const Actions: FunctionComponent = () => <div>ACCIONES</div>;

export { Actions };
