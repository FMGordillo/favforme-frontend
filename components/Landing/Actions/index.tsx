import { FunctionComponent, useEffect, useState } from "react";
import styled from "styled-components";
import { favors as data } from "../../../lib/data";
import { Container, Title } from "../styles";
import { Action } from "./Action";
import { Carousel } from "./Carousel";

const StyledContainer = styled(Container)`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: ${({ theme }) => theme.spacing(2)}em;
`;

// const GET_ACTIONS = gql`
//   {
//     favors {
//       id
//       title
//     }
//   }
// `;

// interface GetActionsData {
//   favors: [Favor];
// }

const Actions: FunctionComponent = () => {
  // const { data } = useQuery<GetActionsData>(GET_ACTIONS);
  const [current, setCurrent] = useState(0);

  useEffect(() => {
    const timer = setTimeout(() => {
      if (!data?.favors.length) return;
      if (current === data.favors.length - 1) {
        setCurrent(0);
      } else {
        setCurrent(current + 1);
      }
    }, 5000);
    return () => {
      clearInterval(timer);
    };
  }, [current]);

  return (
    <StyledContainer id="actions">
      <Title>Acciones Activas</Title>
      <Carousel current={current}>
        {data?.favors.length > 0 &&
          data.favors.map((favor, i) => <Action key={i} data={favor} />)}
      </Carousel>
    </StyledContainer>
  );
};

export { Actions };
