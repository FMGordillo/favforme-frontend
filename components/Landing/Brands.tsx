import Image from "next/image";
import { FunctionComponent } from "react";
import styled from "styled-components";
import { Title } from "./styles";

const Container = styled.section`
  ${({ theme }) => theme.breakpoints.down("md")} {
    margin: 0 ${({ theme }) => theme.spacing(0.5)}em;
  }
`;

const ImagesContainer = styled.div`
  display: flex;
  justify-content: space-evenly;
  & > * {
    margin: ${({ theme }) => theme.spacing(0.1)}em !important;
    max-width: 100%;
    align-self: center;
    text-align: center;
  }
`;

const StyledTitle = styled(Title)`
  text-align: center;
`;

const Brands: FunctionComponent = () => (
  <Container id="brands">
    <StyledTitle>Marcas que nos apoyan</StyledTitle>
    <ImagesContainer>
      <Image src="/images/marcas_01.png" width={186} height={186} />
      <Image src="/images/marcas_02_cut.png" width={126} height={186} />
      <Image src="/images/marcas_03.png" width={186} height={186} />
      <Image src="/images/marcas_04_cut_new.png" width={176} height={146} />
    </ImagesContainer>
  </Container>
);

export { Brands };
