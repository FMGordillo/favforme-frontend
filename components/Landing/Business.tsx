import Image from "next/image";
import { FunctionComponent } from "react";
import styled from "styled-components";
import { Button, Title } from "./styles";

const Container = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const MarginlessTitle = styled(Title)`
  margin-top: 0;
`;

const Subtitle = styled.h2`
  font-weight: 400;
  margin-bottom: 0.5em;
`;

const Business: FunctionComponent = () => {
  return (
    <Container>
      <Subtitle>La oportunidad es ahora</Subtitle>
      <MarginlessTitle>Formá parte del cambio</MarginlessTitle>
      <Image
        src="/images/oficina.png"
        alt="Personas en oficina"
        width={500}
        height={250}
      />
      <p>
        Hacé tu aporte, dando futuro
        <br />a quienes más lo necesitan
      </p>
      <Button color="secondary">Agregar a mi empresa</Button>
    </Container>
  );
};

export { Business };
