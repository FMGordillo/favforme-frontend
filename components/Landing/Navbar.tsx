import Image from "next/image";
import React, { FunctionComponent, useEffect, useState } from "react";
import styled from "styled-components";

const Nav = styled.nav`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 9;
  max-height: 250px; /* From the Logo */
`;
const Ul = styled.ul`
  display: flex;
  justify-content: center;
  list-style: none;
`;
const Item = styled.li`
  flex: 1;
  font-family: dosis, sans-serif;
  text-transform: uppercase;
  font-size: 15px;
  font-weight: 500;
  text-align: center;
  & > a > div {
    max-width: 150px;
  }
`;
const Link = styled.a`
  color: white;
  text-decoration: none;
  &:hover {
    color: ${({ theme }) => theme.color.secondary};
    text-decoration: underline;
  }
`;

const NavItem: FunctionComponent<{ href?: string }> = ({
  children,
  href = "#",
}) => (
  <Item>
    <Link href={href}>{children}</Link>
  </Item>
);

const NavBar: FunctionComponent<{ isScrolled?: boolean }> = ({
  isScrolled = 400,
}) => {
  const [scrollPosition, setScrollPosition] = useState(0);
  const handleScroll = () => {
    const position = window.pageYOffset;
    setScrollPosition(position);
  };

  useEffect(() => {
    if (window) {
      window.addEventListener("scroll", handleScroll, { passive: true });
    }
    return () => {
      if (window) {
        window.removeEventListener("scroll", handleScroll);
      }
    };
  }, []);

  return (
    <Nav>
      <Ul>
        <NavItem>FavForMe</NavItem>
        <NavItem>Empresas</NavItem>
        <NavItem>
          <Image
            src="/images/favforme_logo_white.png"
            width={245}
            height={192}
          />
        </NavItem>
        <NavItem>Marcas</NavItem>
        <NavItem>Contacto</NavItem>
      </Ul>
    </Nav>
  );
};

export { NavBar };
