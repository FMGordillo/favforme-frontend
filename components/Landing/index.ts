export * from "./Business";
export * from "./CallActions";
export * from "./IntroOne";
export * from "./IntroTwo";
