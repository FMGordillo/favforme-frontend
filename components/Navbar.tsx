import Image from "next/image";
import React, { cloneElement, FunctionComponent, useState } from "react";
import styled from "styled-components";

interface ScrolledI {
  isScrolled?: boolean;
  scrollingDown?: boolean;
}

const Nav = styled.nav<ScrolledI>`
  position: absolute;
  left: 0;
  right: 0;
  z-index: 9;
  max-height: 250px; /* From the Logo */
  border-radius: 0 0 100% 100%/0 0 25% 25%;
  transition: all 300ms;

  ${({ theme }) => theme.breakpoints.down("sm")} {
    /* TODO */
  }
`;
const MobileNavigator = styled.div<{ open?: boolean }>`
  display: none;
  position: fixed;
  top: 0;
  z-index: 9;
  height: ${({ theme }) => theme.spacing(1)}em;
  width: 100vw;
  padding: 1em;
  ${({ theme }) => theme.breakpoints.down("sm")} {
    display: block;
  }
`;

const Ul = styled.ul<{ open?: boolean }>`
  display: flex;
  list-style: none;
  padding: 0;
  justify-content: center;
  ${({ theme }) => theme.breakpoints.down("sm")} {
    transform: translateY(-100%);
    background: ${({ theme }) => theme.color.primary.dark};
    flex-direction: column;
    align-items: center;
    padding-top: 80px;
    margin: 0;
    transition: transform 250ms ease-in;
    & > :nth-child(3) {
      display: none;
    }

    ${({ open }) => open && "transform: translateY(0%);"}

    position: fixed;
    width: 100vw;

    & > :nth-child(3) {
      display: none;
    }

    & > * {
      margin: 1em;
    }

    & > :last-child {
      margin-bottom: 2em;
    }
  }
`;
const Item = styled.li<{ image?: boolean }>`
  flex: 1;
  font-family: dosis, sans-serif;
  text-transform: uppercase;
  font-size: 15px;
  font-weight: 500;
  text-align: center;
  ${({ image }) => image && "max-width: 200px;"}
  & > a > div {
    max-width: 150px;
  }
`;
const Link = styled.a`
  font-weight: 500;
  color: white;
  text-decoration: none;
  cursor: pointer;
  border-bottom: 1px solid transparent;
  transition: all 300ms;
  &:hover {
    font-weight: 700;
    color: ${({ theme }) => theme.color.secondary.main};
  }
`;

interface NavItemProps {
  href?: string;
  image?: boolean;
  onClick?: () => void;
}

const NavItem: FunctionComponent<NavItemProps> = ({
  children,
  onClick,
  href,
}) => {
  const props = Object.assign({}, href && { href }, onClick && { onClick });
  return (
    <Item image>{cloneElement(<Link />, { onClick, ...props, children })}</Item>
  );
};

interface NavBarProps {
  isScrolled?: boolean;
  toggleModal?: () => void;
}

const NavBar: FunctionComponent<NavBarProps> = ({
  //isScrolled = 350,
  toggleModal,
}) => {
  const [open, setOpen] = useState(false);
  //const [scrollPosition, setScrollPosition] = useState(0);
  //const [isGoingDown, setIsGoingDown] = useState(true);

  const handleMobileOpen = () => {
    setOpen(!open);
  };

  // const handleScroll = () => {
  //   const position = window.pageYOffset;
  //   if (position > scrollPosition) {
  //     // Going down!
  //     setIsGoingDown(true);
  //   } else {
  //     setIsGoingDown(false);
  //   }
  //   setScrollPosition(position);
  // };

  // useEffect(() => {
  //   if (window) {
  //     window.addEventListener("scroll", handleScroll, { passive: true });
  //   }
  //   return () => {
  //     if (window) {
  //       window.removeEventListener("scroll", handleScroll);
  //     }
  //   };
  // }, []);

  return (
    <Nav>
      <MobileNavigator open={open}>
        <Image
          onClick={handleMobileOpen}
          layout="fixed"
          width={60}
          height={60}
          src="/images/hamburger.png"
        />
      </MobileNavigator>
      <Ul open={open}>
        <NavItem href="#actions">Acciones</NavItem>
        <NavItem href="#call_to_actions">Features</NavItem>
        <NavItem image href="/">
          <Image
            src="/images/favforme_logo_white.png"
            width={245}
            height={192}
          />
        </NavItem>
        <NavItem href="#brands">Alianzas</NavItem>
        <NavItem onClick={toggleModal}>Contacto</NavItem>
      </Ul>
    </Nav>
  );
};

export { NavBar };
