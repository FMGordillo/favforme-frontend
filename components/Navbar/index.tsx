import { FunctionComponent, useContext, useState } from "react";
import { MobileNavigator, Nav, Ul, User } from "./styles";
import { ContactModal } from "@/components/Modal/components";
import Image from "next/image";
import { Logo } from "../assets";
import { ModalContext } from "@/lib/context";
import { NavItem } from "./NavItem";
import { event } from "@/lib/gtag";
import { useRouter } from "next/router";

interface NavBarProps {
  isScrolled?: boolean;
}

const NavBar: FunctionComponent<NavBarProps> = () => {
  const router = useRouter();
  // const { data: session } = useSession();
  const [user, setUser] = useState("");
  const { handleModal } = useContext(ModalContext);
  const [open, setOpen] = useState(false);

  const handleUserClick = async () => {
    // if (session) router.push("/perfil");
    // else signIn();
    // TODO: Make a Provider / hook for it
    const [account] = (await window.ethereum.request({
      method: "eth_requestAccounts",
    })) as string[];
    setUser(`${account.slice(0, 6)}...`);
  };

  const handleMobileOpen = () => {
    setOpen(!open);
  };

  const handleContactClick = () => {
    event({
      action: "click_contacto_navbar",
      category: "contacto",
      value: 1,
    });
    handleModal(ContactModal);
  };

  return (
    <Nav>
      <MobileNavigator open={open}>
        <Image
          onClick={handleMobileOpen}
          layout="fixed"
          width={60}
          height={60}
          src="/images/hamburger.png"
        />
      </MobileNavigator>
      <Ul open={open}>
        <NavItem href="/nosotros">FavForMe</NavItem>
        <NavItem href="/acciones">Acciones</NavItem>
        <NavItem image href="/">
          <Logo />
        </NavItem>
        <NavItem href="/#brands">Empresas</NavItem>
        <NavItem onClick={handleContactClick}>Contacto</NavItem>
        <NavItem isProfileButton onClick={() => router.push("/perfil")} />
        <User onClick={handleUserClick}>
          <div>
            <Image src={"/images/icon_user.svg"} width={75} height={75} />
          </div>
          <p>{user ? user : "Iniciar sesi√≥n"}</p>
        </User>
      </Ul>
    </Nav>
  );
};

export { NavBar };
