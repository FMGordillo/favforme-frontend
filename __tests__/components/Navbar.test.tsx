import { shallow } from "enzyme";
import { NavBar } from "../../components/Landing/Navbar";

let toggleModal;

beforeEach(() => {
  toggleModal = jest.fn();
});

describe("NavBar component", () => {
  it("should render", () => {
    const component = shallow(<NavBar toggleModal={toggleModal} />);
    expect(component).toMatchSnapshot();
  });
});
