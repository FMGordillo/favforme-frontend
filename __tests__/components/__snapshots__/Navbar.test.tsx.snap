// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render 1`] = `ShallowWrapper {}`;
